### DISCUSSION TITLE
SMTP with TLS

### DISCUSSION THREAD

**Discussion by r34ldude** [modoboa/modoboa#2155] [Q&A] [answered]
Hello everyone,

my IMAP(s) working with port 993 and SSL/TLS.  Perfect!
SMTP works with port 587 and STARTTLS. 

How can I get SSL/TLS for SMTP? (Port 465 with SSL/TLS doesnt work.)

Thanks.

**Comment by tonioo** [comment, marked as answer]
It's not enabled by default. You can check at the /etc/postfix/master.cf file if you need to modify this. Note that your end users should not use this port for communication with other users, it's designed for server to server communications.

**Reply by r34ldude** [reply]
Which Port is designed for sever to server communication? TCP 465? I think TCP 25.?

In /etc/postfix/master.cf I already set;
smtpd_use_tls = yes
smtpd_enforce_tls = yes
smtpd_tls_auth_only = yes

-changed my smtpd_tls_key_file and smtpd_tls_cert_file
-changed my smtpd_tls_dh1024_param_file 

smtp_use_tls = yes
smtp_enforce_tls = yes
smtp_tls_auth_only = yes

-changed my smtp_tls_key_file and smtp_tls_cert_file


But couldnt get it work. I try to connect via Outlook with the following settings
IMAP 993 TLS/SSL
SMTP 465 TLS/SSL


Should I change 'smtpd_tls_security_level = may' and 'smtp_tls_security_level = may' to encrypt?
Than should it work with SMTP 465 TLS/SSL?

**Reply by r34ldude** [reply]
If I changed 'smtpd_tls_security_level = may' and 'smtp_tls_security_level = may' to encrypt I couldnt send mails 
on Port 587 with STARTTLS/(or)/TLS/SSL or Port 465 with TLS/SSL.
:/
--EDIT--
IMAP connection is going well, but SMTP doesnt work; https://tools.ietf.org/html/rfc8314#section-7.2
I dont see anything special in the /var/log/mail.log ode mail.err file.

**Reply by tonioo** [reply]
Port 465 is designed for server to server communications. 587 must be used by users. And as I said, the smtps service is not enabled by default. Look at the /etc/postfix/master.cf file.

**Reply by r34ldude** [reply]
Thanks a lot, now I read right. Seconds before your answer I checked open ports and also understand port 465 is not open.

Now I set (removed comment) in /etc/postfix/master.cf for;
smtps     inet  n       -       -       -       -       smtpd
  -o syslog_name=postfix/smtps
  -o smtpd_tls_wrappermode=yes
  -o smtpd_sasl_auth_enable=yes
  -o smtpd_reject_unlisted_recipient=no
  -o smtpd_client_restrictions=$mua_client_restrictions
  -o smtpd_helo_restrictions=$mua_helo_restrictions
  -o smtpd_sender_restrictions=$mua_sender_restrictions
  -o smtpd_recipient_restrictions=
  -o smtpd_relay_restrictions=permit_sasl_authenticated,reject
  -o milter_macro_daemon_name=ORIGINATING

And now port 465 is open and work with SSL/TLS.
(With 'smtp_tls_loglevel = 2' and 'smtpd_tls_loglevel = 2' I can now see the TLS connection also in the mail.log 👍 )

I just understand not right why I should use 587 (I think then also with STARTTLS). But I will read more :)

Thanks a lot men 👍 👍
